File: backend/src/routes/university.routes.ts
import { Router } from 'express'
import { requireAuth } from '../middlewares/auth.middleware'
import {
  listarMaterias, criarMateria, editarMateria, deletarMateria,
  listarGrade, criarAula, deletarAula,
  listarProvas, criarProva, toggleProva, deletarProva,
} from '../controllers/university.controller'

const router = Router()

router.use(requireAuth)

// Matérias
router.get('/subjects',        listarMaterias)
router.post('/subjects',       criarMateria)
router.put('/subjects/:id',    editarMateria)
router.delete('/subjects/:id', deletarMateria)

// Grade horária
router.get('/schedule',        listarGrade)
router.post('/schedule',       criarAula)
router.delete('/schedule/:id', deletarAula)

// Provas
router.get('/exams',           listarProvas)
router.post('/exams',          criarProva)
router.patch('/exams/:id',     toggleProva)
router.delete('/exams/:id',    deletarProva)

export default router

File: backend/internal/university/university_routes.go
package university

import (
	"github.com/gin-gonic/gin"
	"github.com/jackc/pgx/v5/pgxpool"

	"beeplanerofc/backend-go/internal/middleware"
	"beeplanerofc/backend-go/internal/supabaseauth"
)

// RegisterRoutes registra as rotas universitárias no grupo /api/university. Todas exigem login.
func RegisterRoutes(router *gin.RouterGroup, db *pgxpool.Pool, authClient *supabaseauth.Client) {
	subjects := NewSubjectsController(db)
	schedule := NewScheduleController(db)
	exams := NewExamsController(db)

	router.Use(middleware.RequireAuth(db, authClient))

	// Matérias
	router.GET("/subjects", subjects.ListarMaterias)
	router.POST("/subjects", subjects.CriarMateria)
	router.PUT("/subjects/:id", subjects.EditarMateria)
	router.DELETE("/subjects/:id", subjects.DeletarMateria)

	// Grade horária
	router.GET("/schedule", schedule.ListarGrade)
	router.POST("/schedule", schedule.CriarAula)
	router.DELETE("/schedule/:id", schedule.DeletarAula)

	// Provas
	router.GET("/exams", exams.ListarProvas)
	router.POST("/exams", exams.CriarProva)
	router.PATCH("/exams/:id", exams.ToggleProva)
	router.DELETE("/exams/:id", exams.DeletarProva)
}
